Shaily_StataHw1_10:2.do

```diff
@@ -0,0 +1,234 @@
+        
+        if regexm("`f'", "%.+s") local cmd cap format %-`ff' `v'
+        
+        if "`: value label `v''" != "" & "`label'" != "" ///
+            local cmd cap format %-`ff' `v'
+        
+        if "`all'" != "" local cmd cap format %-`ff' `v'
+        
+        `cmd'
+        
+        local ff : format `v'
+        if "`ff'" != "`f'" local vlist `vlist' `v'
+
+    }
+    
+    if "`vlist'" != "" des `vlist'
+	
+sort table 
+order table, before(table21)
+drop table21 //this column is useless
+drop AC //this column has no observations
+
+*now combine columns and rename according to how they appear in PDF
+*could I make any of the below into a loop?
+gen totalpop = (B + C)
+gen cni_obtained = (D+E)
+gen no_cni = (F+G)
+
+gen m_totalpop = (H+I)
+gen m_cni_obtained = (J+K)
+gen m_no_cni = (L+M)
+
+gen f_totalpop = (N+O)
+gen f_cni_obtained = (P+Q)
+gen f_no_cni = (R+S)
+
+gen t_totalpop = (T+U)
+gen t_cni_obtained = (V+W)
+gen t_no_cni = (AA+AB+Z+X+Y)
+
+drop B C D E F G H I J K L M N O P Q R S T U V W X Y Z AA AB
+
+destring(totalpop t_totalpop t_no_cni t_cni_obtained no_cni m_totalpop m_no_cni m_cni_obtained f_totalpop f_no_cni f_cni_obtained cni_obtained), replace
+encode m_no_cni, gen(m_no_cni1)
+encode f_totalpop, gen(f_totalpop1)
+encode f_cni_obtained, gen(f_cni_obtained1)
+encode t_totalpop, gen(t_totalpop1)
+encode t_cni_obtained, gen(t_cni_obtained1)
+
+drop m_no_cni f_totalpop f_cni_obtained t_totalpop t_cni_obtained
+ 
+ren m_no_cni1 m_no_cni
+ren f_totalpop1 f_totalpop
+ren f_cni_obtained1 f_cni_obtained
+ren t_totalpop1 t_totalpop
+ren t_cni_obtained1 t_cni_obtained
+*now, each row represents a district and contains the 12 variables for national id information
+save 18andolder_Pakistan_NID, replace
+
+
+/*******************************************************************************
+Q4: GRANT SCORES
+*******************************************************************************/
+use grant_prop_review_2022.dta, clear
+ren Rewiewer1 Reviewer1
+bysort Reviewer1: egen score1_mean_shaily = mean(Review1Score) if Reviewer1 == "sa1600"
+bysort Reviewer2: egen score2_mean_shaily = mean(Reviewer2Score) if Reviewer2 == "sa1600"
+bysort Reviewer3: egen score3_mean_shaily = mean(Reviewer3Score) if Reviewer3 == "sa1600"
+
+*I spent a lot of time thinking about this problem, and ultimately decided that the best approach would be to see if I could generate the necessary results just based on one reviewer (I chose myselff to be the test case). I was able to generate my own average scores from when I was in the position of all three reviewers, but could not figure out how to average these three scores to generate my actual average score. Once I figure this out, I am hoping to convert this into a loop in order to do the same process to all 16 reviewers, but I also could not figure out how to write this loop since all of the reviewer names are strings and not necessarily in the same order for the 3 reviewer categories. 
+
```

Comment:
Great points and observations. Also, good work on this question as well. 

In this question as well, I found it easier to 

- rename the variable names to ensure uniformity 
- then reshape to a long format to get everyone's names and scores side-by-side in a single column, making it easier to generate reviewer-level z-scores 
- and finally, reshape back to a wide format to calculate average z-scores and rank the proposals. 

There seem to be a lot of approaches to this question based on what I have seen posted here on Github. This was the one I used and found intuitive. I could be wrong, though.